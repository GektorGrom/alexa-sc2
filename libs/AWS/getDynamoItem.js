import AWS from 'aws-sdk';

const parse = AWS.DynamoDB.Converter.unmarshall;

AWS.config.update({
  accessKeyId: process.env.ACCESS_KEY,
  secretAccessKey: process.env.SECRET_ACCESS_KEY,
  region: process.env.AWS_REGION,
});
const client = new AWS.DynamoDB();

const getDynamoItem = async (table = 'SC2_Units', id) => {
  const gameList = await client.getItem({
    TableName: table,
    Key: {
      name_id: {
        S: id,
      },
    },
    ProjectionExpression: 'description',
  }).promise().catch((e) => {
    console.log(e.message);
  });
  return parse(gameList.Item);
};

export default getDynamoItem;
