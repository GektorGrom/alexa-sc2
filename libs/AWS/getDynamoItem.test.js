import getDynamoItem from './getDynamoItem';

test('Get Ghost Item', async (done) => {
  const ghost = await getDynamoItem('SC2_Units', 'Ghost_(Legacy_of_the_Void)');
  expect(ghost).toMatchSnapshot();
  done();
});
