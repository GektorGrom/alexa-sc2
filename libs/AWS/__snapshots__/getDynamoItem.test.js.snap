// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Ghost Item 1`] = `
Object {
  "description": "The Ghost is a specialized infantry unit built from a Barracks with an attached Tech Lab once a Ghost Academy has been constructed. The Ghost stands as an Anti-Spellcaster unit and high damage versus Light units. 
Steady Targeting deals 170 damage regardless of armor to any biological unit at the cost of 50 energy. It cannot be queued to multiple units, so it is mostly effective against high health biological units or for quickly dispatching high impact units. Steady Targeting returns it's energy cost if it fails. 
Steady Targeting one shots Zealots, Dark Templar , High Templar, Roaches , Hydralisks, Zerglings, Infestors, Mutalisks, Marauders , Marines , Ghosts , Reapers, Swarm Hosts, Banelings, Ravagers, Drones , SCVs , and Hellbats.
The following units require two spells to kill: Vipers, Lurkers, Corruptors, Brood Lords, Overseers and Overlords.
An Ultralisk requires three spells to be killed.
The Ghost's Cloak allows it to render itself invisible to enemies without detection. This is generally used to get the Ghost into a good position to EMP or Steady Target an opponent, or to sneak into a base for a Nuke. If using the Cloak to setup an EMP, you must keep an eye on the energy bar as it could prove disastrous if you do not have enough Energy left to cast EMP against various armies.
The EMP Round is the Ghost's most important spell against Protoss due to every unit having shields. Against Zerg and Terran, EMP is effective against dedicated spellcasters. The AoE shot will remove 100 energy (and shields, if applicable) from all units/buildings under its effect. This is essential for use against Protoss opponents for many reasons. The spell effectively does up to 100 damage to anything it hits and can nullify many dangerous Protoss units. Used against a High Templar, the Ghost can prevent the use of Psionic Storm. It also reveals any cloaked units within its radius for a short time, rendering them vulnerable until the effect wears off (though it will not reveal burrowed units). The EMP Round can also be used against structures, where it can damage the shields of Protoss buildings or drain energy from an Orbital Command.
The Tactical Nuke is another ability of the Ghost and will allow the unit to call down a Nuke on a target location provided the Ghost Academy has a Nuke ready and that the Ghost can channel for the full-time. The Nuke is a devastating ability that deals terrible damage to any buildings or units that it hits. However, the cost of using Nukes is very high and requires a heavy amount of gas and minerals to tech to. The long call-down time (14 seconds) also renders it ineffective against units, as they can easily move away. However the large radius of a nuke and possibility to land multiple strikes can allow the Terran to create a more effective push by zoning out areas. The Nuke can also be used as a psychological weapon against your opponent due to it's high damage potential. The Nuke deals 300 full damage to all units in the inner circle of the blast radius,  50% damage to the radius immediately after and 25% damage to the outermost bounds. Note also that the Nuke deals an additional 200 damage to structures, making it highly effective against fortified areas (effectively one shooting all forms of static defense but the planetary fortress. The Nuke is castable from a maximum of 12 range (though the ghost's vision range is only 11). A nuke is not cancelable in the last 0.5-0.7 seconds before impact. During this period the ghost calling the nuke can freely move already.
",
}
`;
